lib/SILOptimizer/Differentiation/JVPCloner.cpp

```diff
@@ -77,6 +77,9 @@ class JVPCloner::Implementation final
   /// tangent values.
   llvm::DenseMap<SILValue, AdjointValue> tangentValueMap;
 
+  enum IsInitialization_t : bool { NotInitialized, Initialized };
```

Comment:
Did you mean `IsInitialized_t`?

That said, I feel the data structure could be simplified to a `llvm::SmallPtrSet<ValueBase *>`, and you can define a method `bool isTangentBufferInitialized(SILValue value)` to access this info.

Reply:
Yes, it should be `IsInitialized_t`.

I think having `map` instead of `set` is a bit safer in the sense that `map` allows us to explicitly track initialization (`true`/`false`) of every observed tangent buffer. This makes it possible to check whether a tangent buffer was previously observed (by `assert(map.count(tanBuf)`), which seems like a good check for correctness/debugging purposes.

Reply:
Keeping `DenseMap` for explicit checking sounds pretty good to me.

Reply:
Ok, sounds good.